lib/eio/dune

```diff
@@ -3,5 +3,4 @@
  (synopsis
   "Storage backend for ambient-context using Eio's fibre-local storage")
  (public_name ambient-context-eio)
- (optional)
```

Comment:
l'oops maybe i misunderstood the `(optional)` flag 😩

Reply:
`optional` is when you have a (sub-)library you might not want to build, in case its deps are not present. For a main library (that someone will `opam install` explicitly, or via transitive deps) it's a bug if its deps are not present.